mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -333,13 +339,18 @@ private PropertyBuilder getPropertyBuilder(Row row) {
         return property;
     }
 
-    private List<AttachmentId> getAttachmentsIds(Row row, FetchType fetchType) {
+    private List<MessageAttachment> getAttachments(Row row, FetchType fetchType) {
         switch (fetchType) {
         case Full:
         case Body:
-            return row.getList(ATTACHMENTS_IDS, String.class)
+            return row.getList(ATTACHMENTS, UDTValue.class)
                     .stream()
-                    .map(AttachmentId::from)
+                    .map(Throwing.function(x -> 
+                        MessageAttachment.builder()
+                            .attachment(attachmentMapper.getAttachment(AttachmentId.from(x.getString(Attachments.ID))))
```

Comment:
you are doing n+1 queries, it looks like a bad idea, you should fetch attachment for n messages in a single query